src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,686 @@
+            return;
+        }
+
+        uint256 unallocatedLiquidity = liquidityBalance - allocatedLiquidity;
+
+        // the new claimable amount is the smaller of the queue shortfall or unallocated weth
+        uint256 addedClaimable = queueShortfall < unallocatedLiquidity ? queueShortfall : unallocatedLiquidity;
+
+        // Store the new claimable amount back to storage
+        withdrawsClaimable = SafeCast.toUint128(withdrawsClaimableMem + addedClaimable);
+    }
+
+    /// @dev Calculate how much of the liquidity asset in the ARM is not reserved for the withdrawal queue.
+    // That is, it is available to be swapped.
+    function _liquidityAvailable() internal view returns (uint256) {
+        // The amount of WETH that is still to be claimed in the withdrawal queue
+        uint256 outstandingWithdrawals = withdrawsQueued - withdrawsClaimed;
+
+        // The amount of the liquidity asset is in the ARM
+        uint256 liquidityBalance = IERC20(liquidityAsset).balanceOf(address(this));
+
+        // If there is not enough liquidity assets in the ARM to cover the outstanding withdrawals
+        if (liquidityBalance <= outstandingWithdrawals) {
+            return 0;
+        }
+
+        return liquidityBalance - outstandingWithdrawals;
+    }
+
+    /// @notice The total amount of assets in the ARM and external withdrawal queue,
+    /// less the liquidity assets reserved for the ARM's withdrawal queue and accrued fees.
+    function totalAssets() public view virtual returns (uint256) {
+        uint256 totalAssetsBeforeFees = _rawTotalAssets();
+
+        // If the total assets have decreased, then we don't charge a performance fee
+        if (totalAssetsBeforeFees <= lastTotalAssets) return totalAssetsBeforeFees;
+
+        // Calculate the increase in assets since the last time fees were calculated
+        uint256 assetIncrease = totalAssetsBeforeFees - lastTotalAssets;
+
+        // Calculate the performance fee and remove from the total assets before new fees are removed
+        return totalAssetsBeforeFees - ((assetIncrease * fee) / FEE_SCALE);
+    }
+
+    /// @dev Calculate the total assets in the ARM, external withdrawal queue,
+    /// less liquidity assets reserved for the ARM's withdrawal queue and past accrued fees.
+    /// The accrued fees are from the last time fees were calculated.
+    function _rawTotalAssets() internal view returns (uint256) {
+        // Get the assets in the ARM and external withdrawal queue
+        uint256 assets = token0.balanceOf(address(this)) + token1.balanceOf(address(this)) + _externalWithdrawQueue();
+
+        // Load the queue metadata from storage into memory
+        uint256 queuedMem = withdrawsQueued;
+        uint256 claimedMem = withdrawsClaimed;
+
+        // If the ARM becomes insolvent enough that the total value in the ARM and external withdrawal queue
+        // is less than the outstanding withdrawals.
+        if (assets + claimedMem < queuedMem) {
```

Comment:
I don't think this check actually checks the entire subtraction that follows? The bottom one has an additional - `feesAccrued`?

Reply:
Or is reverting intentional.

Reply:
This is a bug. It should return 0 rather than reverting. I've fixed